src/MagnumPlugins/DrFlacAudioImporter/DrFlacImporter.cpp

```diff
@@ -74,12 +103,30 @@ void DrFlacImporter::doOpenData(Containers::ArrayView<const char> data) {
         return;
     }
 
-    Containers::Array<char> tempData(samples*sizeof(int32_t));
+    _frequency = handle->sampleRate;
+    _format = flacFormatTable[numChannels][normalizedBitsPerSample];
 
-    drflac_read_s32(handle, samples, reinterpret_cast<int32_t*>(tempData.begin()));
+    Containers::Array<char> tempData(samples*sizeof(Int));
+    drflac_read_s32(handle, samples, reinterpret_cast<Int*>(tempData.begin()));
     drflac_close(handle);
 
-    _data = std::move(tempData);
+    switch(normalizedBitsPerSample)
+    {
+        case 1: {
+            _data = convert32PCM(tempData, samples, sizeof(UnsignedByte));
+
+            // Convert to unsigned
+            for(char& item : _data) {
+                item = item - 128;
+            }
+            break;
+        }
+        case 2: {
+            _data = convert32PCM(tempData, samples, sizeof(Short));
+            break;
+        }
+    }
```

Comment:
@mosra It won't be shorter once I have 3 (24 bit audio), 4 (32 bit audio), and 8 (64 bit audio) working.

convert32PCM() drops a configurable amount of data; DrFlac loads EVERYTHING into 32 bit PCM by zero padding, so we drop the padding to recover the original data.